oximeter/db/src/client.rs

```diff
@@ -2957,105 +2959,156 @@ mod tests {
         let datum = Datum::U16(1);
-        let as_json = serde_json::Value::from(1_u16);
-        test_recall_measurement_impl::<u8>(datum, None, as_json, client)
+        let measurement = Measurement::new(Utc::now(), datum);
+        test_recall_measurement_impl(measurement.clone(), client).await?;
+        test_recall_missing_scalar_measurement_impl(measurement, client)
             .await?;
         Ok(())
     }
 
     async fn recall_measurement_i32_test(client: &Client) -> Result<(), Error> {
         let datum = Datum::I32(1);
-        let as_json = serde_json::Value::from(1_i32);
-        test_recall_measurement_impl::<u8>(datum, None, as_json, client)
+        let measurement = Measurement::new(Utc::now(), datum);
+        test_recall_measurement_impl(measurement.clone(), client).await?;
+        test_recall_missing_scalar_measurement_impl(measurement, client)
             .await?;
         Ok(())
     }
 
     async fn recall_measurement_u32_test(client: &Client) -> Result<(), Error> {
         let datum = Datum::U32(1);
-        let as_json = serde_json::Value::from(1_u32);
-        test_recall_measurement_impl::<u8>(datum, None, as_json, client)
+        let measurement = Measurement::new(Utc::now(), datum);
+        test_recall_measurement_impl(measurement.clone(), client).await?;
+        test_recall_missing_scalar_measurement_impl(measurement, client)
             .await?;
         Ok(())
     }
 
     async fn recall_measurement_i64_test(client: &Client) -> Result<(), Error> {
         let datum = Datum::I64(1);
-        let as_json = serde_json::Value::from(1_i64);
-        test_recall_measurement_impl::<u8>(datum, None, as_json, client)
+        let measurement = Measurement::new(Utc::now(), datum);
+        test_recall_measurement_impl(measurement.clone(), client).await?;
+        test_recall_missing_scalar_measurement_impl(measurement, client)
             .await?;
         Ok(())
     }
 
     async fn recall_measurement_u64_test(client: &Client) -> Result<(), Error> {
         let datum = Datum::U64(1);
-        let as_json = serde_json::Value::from(1_u64);
-        test_recall_measurement_impl::<u8>(datum, None, as_json, client)
+        let measurement = Measurement::new(Utc::now(), datum);
+        test_recall_measurement_impl(measurement.clone(), client).await?;
+        test_recall_missing_scalar_measurement_impl(measurement, client)
             .await?;
         Ok(())
     }
 
     async fn recall_measurement_f32_test(client: &Client) -> Result<(), Error> {
         const VALUE: f32 = 1.1;
         let datum = Datum::F32(VALUE);
-        // NOTE: This is intentionally an f64.
-        let as_json = serde_json::Value::from(1.1_f64);
-        test_recall_measurement_impl::<u8>(datum, None, as_json, client)
+        let measurement = Measurement::new(Utc::now(), datum);
+        test_recall_measurement_impl(measurement.clone(), client).await?;
+        test_recall_missing_scalar_measurement_impl(measurement, client)
```

Comment:
Out of curiosity, when did the restriction that this should be an f64 get lifted?